tests/p2p/test_protocol.py

```diff
@@ -176,7 +178,8 @@ def test_invalid_peer_id(self):
         # So we need to stop this call manually, otherwise the reactor would be unclean with a pending call
         # TODO We should use a fake DNS resolver for tests otherwise we would need internet connection to run it
         resolver = twisted.names.client.getResolver().resolvers[2]
-        resolver._parseCall.cancel()
+        if not resolver._parseCall.cancelled:
```

Comment:
This is very old. I think we already have a method that properly cleans up the pending calls.

Reply:
OK, I'll remove this along the `skipif`

Reply:
Great!